File: leet_BFS_0429_nary_tree_level_order___bfs2.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root)
    {
        vector<vector<int>> r;
        deque<Node*>        dq;

        if (!root)
            return r;
        dq.push_back(root);
        r.push_back({root->val}); // added
        while (!dq.empty())
        {
            int len = dq.size();
            vector<int> temp;

            while (len--)
            {
                Node *node = dq.front();

                dq.pop_front();
                for (Node *n: node->children)
                {
                    dq.push_back(n);
                    temp.push_back(n->val); // added
                }
            }
            if (!temp.empty())
                r.push_back(temp);
        }
        return r;
    }
};

File: leet_BFS_0429_nary_tree_level_order___bfs2.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */

var levelOrder = function(root) {
    let r = []
    if (!root) {
        return r
    }
    r.push([root.val])
    let dq = [root]
    while (dq.length > 0) {
        let temp = []
        let len = dq.length
        while (len-- > 0) {
            let node = dq.shift()
            for (let n of node.children) {
                dq.push(n)
                temp.push(n.val)
            }
        }
        if (temp.length > 0) {
            r.push(temp)
        }
    }
    return r
};
